onnxruntime/core/optimizer/bias_dropout_fusion.cc

```diff
@@ -15,49 +15,56 @@ void FuseResidualAddIfAny(Graph& graph, const Node& dropout_node,
                           std::vector<NodeArg*>& dropout_output,
                           std::vector<std::reference_wrapper<Node>>& nodes_to_fuse) {
   bool has_residual_add = false;
-  for (auto last_node_itr = dropout_node.OutputNodesBegin(); last_node_itr != dropout_node.OutputNodesEnd(); ++last_node_itr) {
-    const Node& last_node = (*last_node_itr);
-
-    if (graph_utils::IsSupportedOptypeVersionAndDomain(last_node, "Add", {7, 13}) &&
-        last_node.GetExecutionProviderType() == dropout_node.GetExecutionProviderType()) {
-      const TensorShapeProto* input1_shape = last_node.InputDefs()[0]->Shape();
-      const TensorShapeProto* input2_shape = last_node.InputDefs()[1]->Shape();
-
-      if (input1_shape == nullptr ||
-          input2_shape == nullptr ||
-          input1_shape->dim_size() < 1 ||
-          input2_shape->dim_size() < 1 ||
-          input1_shape->dim_size() != input2_shape->dim_size()) {
-        continue;
-      }
-
-      // Inputs of Residual Add must match in shape
-      bool match = true;
-      for (int i = 0; i < input1_shape->dim_size(); ++i) {
-        match &= ONNX_NAMESPACE::operator==(input1_shape->dim(i), input2_shape->dim(i));
-      }
-      if (!match) {
-        continue;
-      }
-
-      // dropout's output is not part of of graph output
-      if (!graph.GetNodeOutputsInGraphOutputs(dropout_node).empty()) {
-        continue;
-      }
 
-      Node& residual_add_node = *graph.GetNode(last_node.Index());
-      const std::string& dropout_output_name = dropout_node.OutputDefs()[0]->Name();
-      if (dropout_output_name == residual_add_node.InputDefs()[0]->Name()) {
-        dropout_input.push_back(residual_add_node.MutableInputDefs()[1]);  // residual
-      } else if (dropout_output_name == residual_add_node.InputDefs()[1]->Name()) {
-        dropout_input.push_back(residual_add_node.MutableInputDefs()[0]);  // residual
+  int dropout_consumers_count = 0;
+  for (auto edge_itr = dropout_node.OutputEdgesBegin(); edge_itr != dropout_node.OutputEdgesEnd(); ++edge_itr) {
```

Comment:
you can use dropout_node.GetOutputEdgesCount() instead

Reply:
`GetOutputEdgesCount()` will count all edges, even for the mask output, right?
I only need edges for the output 0.

Reply:
got it